exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinProbeTemplate.java

```diff
@@ -262,6 +272,7 @@ private void executeProbePhase() throws SchemaChangeException {
                 probeBatch.getSchema());
             }
           case OK:
+            setTargetOutputCount(outgoingJoinBatch.getBatchMemoryManager().update(probeBatch, LEFT_INDEX,outputRecords));
```

Comment:
It will not make it bigger. It will look at remaining memory and adjust the row count based on that. 

    final long remainingMemory = Math.max(configOutputBatchSize - memoryUsed, 0);
    // These are number of rows we can fit in remaining memory based on new outgoing row width.
    final int numOutputRowsRemaining = RecordBatchSizer.safeDivide(remainingMemory, newOutgoingRowWidth);